core/src/main/java/org/apache/calcite/adapter/enumerable/EnumUtils.java

```diff
@@ -354,6 +354,9 @@ public static Expression convert(Expression operand, Type fromType,
     if (!Types.needTypeCast(fromType, toType)) {
       return operand;
     }
+
+    // TODO consider using Expressions#convertChecked to throw exception in case of overflow
```

Comment:
@rubenada we might want to file a ticket for this and replace `TODO` with the ticket reference